model/Service/Session/EncryptedUser.php

```diff
@@ -50,8 +50,10 @@ public function __construct(common_user_User $user, $hashForEncryption = null)
         $this->realUser = $user;
 
         $password = $this->realUser->getPropertyValues(GenerisRdf::PROPERTY_USER_PASSWORD);
-        $salt     = $password[0];
-        $this->key = GenerateKey::generate($hashForEncryption, $salt);
+        if (isset($password[0])){
+            $salt     = $password[0];
+            $this->key = GenerateKey::generate($hashForEncryption, $salt);
+        }
```

Comment:
else Exception?

Reply:
Not absolute necessary, if no password you are not able to login. This is a check to avoid having undefined index warning.